### ISSUE TITLE
Error threaded_load-ing two scenes that inherit from the same scene

### ISSUE THREAD

**Issue by zzatuchn** [godotengine/godot#66023]
### Godot version

4.0beta1

### System information

Windows 10 Home 19044.2006, AMD Radeon RX 580 Series, Driver Version 22.20.19.09-220819a-383490E-AMD-Software-Adrenalin-Edition, Vulkan Clustered

### Issue description

Let's say I have a base scene `scene1.tscn` and two scenes `scene2.tscn` and `scene3.tscn` that inherit from the first. I load_threaded_request them on the same frame and check if they're finished in `_process()` with `load_threaded_get_status()`. `scene2.tscn` loads fine but `scene3.tscn` gives the following error:

```
ERROR: res://scene3.tscn:3 - Parse Error: [ext_resource] referenced nonexistent resource at: res://scene1.tscn
   at: load (scene/resources/resource_format_text.cpp:486)
ERROR: Failed loading resource: res://scene3.tscn. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:221)
```

### Steps to reproduce

Run the minimal example in the editor.

main.gd:

```gdscript
extends Node2D

var x = false
var y = false

func _ready():
	var err = ResourceLoader.load_threaded_request("res://scene2.tscn")
	assert(!err)
	err = ResourceLoader.load_threaded_request("res://scene3.tscn")
	assert(!err)

func _process(delta):
	if !x:
		var status = ResourceLoader.load_threaded_get_status("res://scene2.tscn")
		match status:
			0:
				print("invalid resource res://scene2.tscn")
				x = true
			1:
				pass
			2:
				print("error loading res://scene2.tscn")
				x = true
			3:
				print("finished loading res://scene2.tscn")
				var sc2 = ResourceLoader.load_threaded_get("res://scene2.tscn").instantiate()
				sc2.queue_free()
				x = true
	
	if !y:
		var status = ResourceLoader.load_threaded_get_status("res://scene3.tscn")
		match status:
			0:
				print("invalid resource res://scene3.tscn")
				y = true
			1:
				pass
			2:
				print("error loading res://scene3.tscn")
				y = true
			3:
				print("finished loading res://scene3.tscn")
				var sc3 = ResourceLoader.load_threaded_get("res://scene3.tscn").instantiate()
				sc3.queue_free()
				y = true
```

### Minimal reproduction project

[Resource Load Test.zip](https://github.com/godotengine/godot/files/9592946/Resource.Load.Test.zip)

**Comment by gotnospirit** [comment]
The problem doesn't occur when using `use_sub_threads=true`.

https://github.com/godotengine/godot/blob/24115beb3c69e648b1d48f969785b9b0729e1be8/core/io/resource_loader.cpp#L566-L568
It also doesn't occur when swapping these two lines, but not sure if it's the right correction

**Comment by tinmanjuggernaut** [comment]
Maybe a duplicate of #62159.

**Comment by akien-mga** [comment]
Fixed by #73647.
